### PR TITLE
Create GenericStackList.JAVA

### PR DESCRIPTION
**Pull request by nogsp**
### **Describe your change:**

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**

* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Java files are placed inside an existing directory.
* [] All filenames are in all uppercase characters with no spaces or dashes.
* [X] All functions and variable names follow Java naming conventions.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by P-N-C** [comment]
Posted a Generic Stack implemented by using Linked List

File: DataStructures/Stacks/GenericStackList.JAVA [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		public Link(E it, Link nextval) {
+			element = it;
+			next = nextval;
+		}
+		public Link(Link nextval) {
+			next = nextval;
+		}
+	}
+	
+	public Link create_link(E it, Link nextval) {
```
Comment (aravinth2094) [inline DataStructures/Stacks/GenericStackList.JAVA:65]: Please follow java conventions for a more standard experience.
```suggestion
	public Link createLink(E it, Link nextVal) {
```

File: DataStructures/Stacks/GenericStackList.JAVA [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	
+	public E top() {
+		if(head.next == null) return null;
+		return head.next.element;
+	}
+	
+	public void printStack() {
+		Link temp = head;
+		while(temp.next != null) {
+			if(temp == head) System.out.print(temp.next.element);
```
Comment (aravinth2094) [inline DataStructures/Stacks/GenericStackList.JAVA:46]: Please use getters/setters as it promotes the use of abstraction in java.
```suggestion
			if(temp == head) System.out.print(temp.getNext().getElement());
```

File: DataStructures/Stacks/GenericStackList.JAVA [hunk 3]
```diff
@@ -0,0 +1,75 @@
+public class Stack <E> {
+	
+	Link head;
+	int cnt;
```
Comment (aravinth2094) [inline DataStructures/Stacks/GenericStackList.JAVA:4]: Please consider adopting a more comprehensive nomenclature.
```suggestion
	int count;
```

File: DataStructures/Stacks/GenericStackList.JAVA [hunk 4]
```diff
@@ -0,0 +1,75 @@
+public class Stack <E> {
```
Comment (aravinth2094) [inline DataStructures/Stacks/GenericStackList.JAVA:1]: I highly encourage documentation for the class and its methods.

**Review by aravinth2094 — CHANGES_REQUESTED** [review]

**Comment by P-N-C** [comment]
Thank you, I am going to update my code once my college exams ends.

**Comment by aravinth2094** [comment]
> Thank you, I am going to update my code once my college exams ends.

Great, All the best!

**Review by ayaankhan98 — COMMENTED** [review]
also please add the relevant documentation and rename file to change extension from `.JAVA` to `.java` whenever you update this PR.

**Pull request closed without merging**